packages/connector-jsdom/src/connector.ts

```diff
@@ -167,6 +128,21 @@ export default class JSDOMConnector implements IConnector {
         }
     }
 
+    private parseHTML(): Promise<void> {
+        return new Promise(async (resolve) => {
+            const html = await this.html;
+
+            await this.server.on('end-traverse::html', () => {
+                resolve();
+            });
+
+            await this.server.emitAsync('can-traverse::html', {
```

Comment:
We had a conversation about this but I can't remember now why can't we use `fetch::end::html`. Can you please explain it here?

Reply:
fetch::end::html is trigger at the begining, when you are fetching the site, but we don't want to traverse that html, we need to wait (`waitFor`) until the scripts have run and then, traverse the html. That is why we can't use the event `fetch::end::html`

Reply:
Also, I think we should have further discussion about whether we want `fetch::end::html` to still trigger parsing, just not traversal.

Context: I'd like to use `parse::end::html` in the future to analyze HTML beyond the top-level page (e.g. nested frames or fragments fetched via XHR), but with the expectation that hints using this are directly interrogating the `IAsyncWindow` provided by `parse::end::html` instead of listening for traversal/element events.

Reply:
`hint-meta-charset-utf-8` requires the initial HTML to validate the tag is in the right place. Right now it's using cheerio and the initial HTML. This could be a good opportunity to rethink how/when we are doing all the HTML parsing.

Reply:
What about having bot events `fetch::end::html` will just parse the html and will return `parse::end::html`.
Then we can have `can-traverse::html` which will use a new jsdom object to traverse.

Reply:
I'm thinking `parse::end::html` should always represent the initial HTML (allowing for static analysis of the original source) and the traversal events should always represent the "initialized" DOM (dynamic analysis). Hints should use one or the other depending on their needs (if a hint uses both it will need to ensure it avoids duplicate reports).

To me that means when we're taking a snapshot of the DOM, we should NOT generate `parse::end::html`, but we should generate it in response to `fetch::end::html` events.

Reply:
Let me clarify this, because I think we are thinking the same hehehe.
`fetch::end::html` will end emitting a `parse::end::html` and will not traverse the html.
`can-traverse::html` will traverse the html that it get by param and will NOT emit `parse::end::html`.

Is this what are you saying?

Reply:
Yep, I think we're on the same page as I didn't see your comment until after I posted mine. 😉

Even so, I'll paraphrase just to lock it in. 
1. `parser-html` will continue respond to the existing `fetch::end::html` event and generate a `parse::end::html` event that includes an `IAsyncWindow` generated via `jsdom`, but will NOT traverse the HTML in this context.
2. `parser-html` will additionally respond to the new `can-traverse::html` event to traverse the HTML using `jsdom`, but will NOT generate a `parse::end::html` in this context.

Reply:
Perfect! I will work on that.

Reply:
Small clarification because it got me thinking for a minute:

2. `parser-html` will additionally respond to the new `can-traverse::html` event to traverse the HTML using `jsdom` sending all the `traverse` events (like `traverse::end`) but will NOT generate a `parse::end::html` in this context.